sdk/include/opentelemetry/sdk/trace/sampler.h

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include "opentelemetry/common/attribute_value.h"
+#include "opentelemetry/trace/span.h"
+#include "opentelemetry/trace/trace_id.h"
+#include "opentelemetry/version.h"
+
+#include <map>
+#include <memory>
+#include <string>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace trace
+{
+namespace trace_api = opentelemetry::trace;
+
+enum class Decision
+{
+  // IsRecording() == false, span will not be recorded and all events and attributes will be
+  //      dropped.
+  NOT_RECORD,
+  // IsRecording() == true, but Sampled flag MUST NOT be set.
+  RECORD,
+  // IsRecording() == true AND Sampled flag` MUST be set.
+  RECORD_AND_SAMPLE
+};
+
+struct SamplingResult
+{
+  Decision decision;
+  // A set of span Attributes that will also be added to the Span
+  std::unique_ptr<std::map<std::string, openelemetry::common::AttributeValue>> attributes;
+};
+
+class Sampler
```

Comment:
It would be great if you could add a short docstring for each of `Sampler`, `SamplingResult` and `Decisision`, like [here](https://github.com/open-telemetry/opentelemetry-cpp/blob/master/sdk/include/opentelemetry/sdk/trace/processor.h#L13).